src/Chapter1.hs

```diff
@@ -653,7 +653,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Integral a => a -> a
+firstDigit n
+    | abs n > 9 = firstDigit (abs n `div` 10)
+    | otherwise = abs n
```

Comment:
Here you call `abs` on each recursive call. You can implement a helper recursive function and call `abs` only once in the beginning :slightly_smiling_face: This should make the implementation a bit more performant :racehorse: